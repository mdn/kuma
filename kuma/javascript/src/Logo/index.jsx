// @flow
import * as React from 'react';

const ducksStyle = Object.freeze({
    font: 'bold 26px/29px zillaslab',
    backgroundColor: 'black',
    color: 'white',
    position: 'absolute',
    width: 'calc(219px - 50px)',
    height: '29px',
    top: '15px',
    marginLeft: '50px',
    zIndex: '100',
});

const h2Style = Object.freeze({
    position: 'absolute',
    height: '48px',
    width: '48px',
    lineHeight: '55px',
    left: '11px',
    top: '-5px',
    zIndex: 100,
    backgroundColor: 'black',
    transform: 'scale(-1, 1)',
});

export default ({url} : {|url: string|}) : React.Node => <div>
    <a href={url} className="logo">MDN Web Docs</a>
    <div style={ducksStyle}>ducks&nbsp;</div>
    <h2 style={h2Style}>&#8239;&#x1f986;&#8239;</h2>
</div>;
