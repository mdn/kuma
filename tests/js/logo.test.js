import React from 'react';
import { create } from 'react-test-renderer';
import Logo from '../../kuma/javascript/src/logo.jsx';

test('Logo snapshot', () => {
    const logo = create(<Logo url="https://developer.mozilla.org" />);
    expect(logo.toJSON()).toMatchSnapshot();
});
