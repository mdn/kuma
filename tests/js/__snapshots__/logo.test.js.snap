// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo snapshot 1`] = `
<a
  css={
    Object {
      "display": "block",
      "height": 48,
      "marginTop": 15,
      "overflow": "hidden",
      "width": 219,
    }
  }
  href="https://developer.mozilla.org"
>
  <img
    alt="MDN Web Docs"
    src="/static/img/web-docs-sprite.svg"
  />
</a>
`;
